### ISSUE TITLE
GPS-Net: Graph Property Sensing Network for Scene Graph Generation

### ISSUE THREAD

**Issue by SerialLain3170** [SerialLain3170/nagato#26]
## Overview

- Authors proposed three modules
  - Direction-aware Message Passing for edge direction
  - Node Priority Sensitive Loss for node priority
  - Adaptive Reasoning Module for long-tailed class distribution

<img width="1009" alt="Screen Shot 2021-08-22 at 13 46 39" src="https://user-images.githubusercontent.com/32360147/130343665-7f54bf34-a63a-432c-8c49-24a4e5194f16.png">

- Direction-aware Message Passing
  - Using tri-linear module, authors employ contextual information with union box features. Moreover, in order to deal with the edge direction, authors use both forward and backward contextual coefficient.
  - To refine contextual information, authors adopt transformer layer.

<img width="1026" alt="Screen Shot 2021-08-22 at 14 20 20" src="https://user-images.githubusercontent.com/32360147/130343671-02776900-cc56-41af-acae-3ff2839024b9.png">

- Node Priority Sensitive Loss
  - Based on focal loss, authors use loss function considering node priority. Like the figure above, low node priority becomes more sensitive.

<img width="506" alt="Screen Shot 2021-08-22 at 14 26 03" src="https://user-images.githubusercontent.com/32360147/130343673-87b959e7-5965-4699-acb0-2f8c4900a56d.png">

- Adaptive Reasoning Module
  - Authors adopt a log-softmax to soften the original class distribution.

**Comment by SerialLain3170** [comment]
## Evaluation

- Authors use three metrics
  - Predicate Classification
  - Scene Graph Classification
  - Scene Graph Detection

- Authors show that proposed method is competitive to previous methods.

<img width="1017" alt="Screen Shot 2021-08-22 at 14 31 02" src="https://user-images.githubusercontent.com/32360147/130343895-bb4c3ac6-083f-458a-82f8-c257043ff813.png">

- Authors also show the effectiveness of each proposed module.

<img width="1017" alt="Screen Shot 2021-08-22 at 14 31 02" src="https://user-images.githubusercontent.com/32360147/130343925-0440a3cf-0353-4df0-8265-50d474bebb3b.png">

**Comment by SerialLain3170** [comment]
## Supplemental Information

[Paper Link](https://openaccess.thecvf.com/content_CVPR_2020/papers/Lin_GPS-Net_Graph_Property_Sensing_Network_for_Scene_Graph_Generation_CVPR_2020_paper.pdf)

### Authors
Xin Lin, Changxing Ding, Jinquan Zeng, Dacheng Tao

### Organization
School of Electronic and Information Engineering, South China University of Technology
UBTECH Sydney AI Centre, School of Computer Science, Faculty of Engineering,
The University of Sydney, Darlington, NSW 2008, Australia

### Conference
CVPR 2020
